flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/AbstractDataOutput.java

```diff
@@ -35,17 +35,12 @@
 	/** The maintainer toggles the current stream status. */
```

Comment:
Maybe you could state in commit message, why the lock is unnecessary here (old issue that has been solved?)